### PR TITLE
PUB-1056 - Rejection reasons

### PR DESCRIPTION
**Pull request by furnivall**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PUB-1056

### Change description ###
Added template and such for media reject email

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/PersonalisationService.java [hunk 1]
```diff
@@ -66,6 +69,7 @@ public class PersonalisationService {
     @Value("${file-retention-weeks}")
     private String fileRetentionWeeks;
 
+    private static final String REJECT_REASONS = "reject_reasons";
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/PersonalisationService.java:72]: Think there is a problem here. The reject reasons placeholder on the email template is "reject-reasons" with a hyphen not underscore. The same for full name. On the email template it is "full-name" but you have "full_name" here. If you just update the e-mail template it should be fine

File: src/test/java/uk/gov/hmcts/reform/pip/publication/services/service/EmailServiceTest.java [hunk 2]
```diff
@@ -262,11 +283,43 @@ void testMediaVerificationEmailReturnsSuccess() {
...
-                     TEMPLATE_MESSAGE);
+                     TEMPLATE_MESSAGE
+        );
+    }
+
+    @Test
+    void testBuildMediaApplicationRejectionEmail() {
+        MediaRejectionEmail mediaRejectionEmail = new MediaRejectionEmail(
+            "John Doe",
+            "sausage@fun.com",
```
Comment (ChrisS1512) [inline src/test/java/uk/gov/hmcts/reform/pip/publication/services/service/EmailServiceTest.java:300]: Email's need updating (and I'd change the name as well to just Test Name or something while you're there)

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/notify/Templates.java [hunk 3]
```diff
@@ -13,6 +13,7 @@ public enum Templates {
     MEDIA_APPLICATION_REPORTING_EMAIL("c59c90a3-1806-4649-b4b5-b6bce8f8f72c"),
     BAD_BLOB_EMAIL("0fbd150f-ff5b-49f0-aa34-6a6273901ceb"),
     MEDIA_USER_VERIFICATION_EMAIL("1dea6b4b-48b6-4eb1-8b86-7031de5502d9"),
+    MEDIA_USER_REJECTION_EMAIL("1988bbdd-d223-49bf-912f-ed34cb43e35e"),
```
[the remark this answers is not in the document]
Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/notify/Templates.java:16]: Sorry i just copied it over. Are we allowed to put links in notify things?

**Review by KianKwa — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/controllers/NotificationController.java [hunk 4]
```diff
@@ -180,6 +182,21 @@ public ResponseEntity<String> sendMediaUserVerificationEmail(@RequestBody MediaV
         ));
     }
 
+    @ApiResponse(responseCode = OK_RESPONSE, description = "Media user rejection email successfully "
+        + "sent with referenceId: {Id}")
+    @ApiResponse(responseCode = BAD_REQUEST, description = BAD_PAYLOAD_ERROR_MESSAGE)
+    @ApiResponse(responseCode = BAD_REQUEST, description = NOTIFY_EXCEPTION_ERROR_MESSAGE)
+    @ApiResponse(responseCode = AUTH_RESPONSE, description = NOT_AUTHORIZED_MESSAGE)
+    @Operation(summary = "Send a media applicant a rejection email")
+    @PostMapping("/media-account/reject")
```
Comment (ashwini-mv) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/controllers/NotificationController.java:191]: Would  renaming it to "/media/reject" make it consistent with other end point urls?

Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/controllers/NotificationController.java]: Updated :)

**Review by KianKwa — APPROVED** [review]

**Review by joshblackmoor — APPROVED** [review]

**Review by joshblackmoor — APPROVED** [review]

**Pull request merged**